### ISSUE TITLE
Titles may show "Missing Datasource" incorectly

### ISSUE THREAD

**Issue by KnoAll** [Cacti/cacti#4457]
I'm seeing a log warning about missing data sources with a link to the data source, which when I click there is an error that it does not exist.
![image](https://user-images.githubusercontent.com/7632685/140585188-065d8ea1-c92f-4de8-835a-072e1e50af35.png)
![image](https://user-images.githubusercontent.com/7632685/140585220-6779d7af-100a-469d-84c3-c7a557bb5d2e.png)

Cacti v 1.2.19

CentOS7

I've tried correlating graphs to the DS but have had no luck

**Comment by netniV** [comment]
When you are viewing the log, does it not have anything else on the log line?

**Comment by KnoAll** [comment]
No, just that entry.
![image](https://user-images.githubusercontent.com/7632685/140785657-4bd9e4a0-2a65-4c1e-a2a5-190037d6d435.png)

this is on Low logging, i can bump that up as needed

**Comment by netniV** [comment]
That's strange that it would be trying to poll for a datasource that doesn't even exist.  The code is basically looking for a title for the datasource in `get_data_source_title()` and failing to do so.  Are these untemplated sources?  If you browse to https://ubuntu1804.netniv.com/data_sources.php?action=ds_edit&id= (just add the ID to the end) can you edit these sources?

**Comment by netniV** [comment]
I've made a slight change to the code, see if that helps you out.

**Comment by KnoAll** [comment]
Sorry for the late response, I didn't get a notification of your comments. I'll take a look...

**Comment by KnoAll** [comment]
I tried directly editing one of the missing DS directly as you indicated and got the same message, does not exist.
I have not yet had a chance to grab the new code...

**Comment by KnoAll** [comment]
Getting the new functions.php did not change anything? Does it add logging or just handle things differently?

**Comment by netniV** [comment]
The only thing that the changes I made did was to allow a non-templated title not become an issue, and identify if a linked template was missing.

You may not notice a difference especially if the datasource in question is actually missing.  You may need to find the graphs that those are associated to.

**Comment by TheWitness** [comment]
```sql
SELECT * FROM data_local WHERE id = ?
SELECT * FROM data_template_data WHERE local_data_id = ?
SELECT * FROM poller_item WHERE local_data_id = ?
```

Are there any entries there?

**Comment by TheWitness** [comment]
I'm pretty sure this should be fixed in the 1.2.x branch.  As far as your damaged data source, you may need to remove the entry from the tables `data_local` and `data_template_data`, and `data_template_rrd`, and then run the `repair_database.php` script.
